Fix missing emitter fields of `Initial Stretch X` and `Initial Stretch Y` and `Initial Angular Velocity` for multiple emitters

wookieejedi opened the pull request:
Adds emitter fields of `Initial Stretch X` and `Initial Stretch Y` and `Initial Angular Velocity` to emitters that added after a particle effects that use more then one emitter. Prviously these fields were missing from `template.emitter` which caused these fields to be greyed out for particle effects that used more than one emitter. Also clicking on those fields and adding a value causes an error `java.lang.NullPointerException: Unknown`.

Fixes #7342 
Fixes #5346

## PR checklist

* [X] Code
	* [X] Add engine and/or editor unit tests.
	* [X] New and changed code follows the overall code style of existing code
	* [X] Add comments where needed
* [X] Documentation
	* [X] Make sure that API documentation is updated in code comments
	* [X] Make sure that manuals are updated (in github.com/defold/doc)
* [X] Prepare pull request and affected issue for automatic release notes generator
	* [X] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [X] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [X] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [X] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [X] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [X] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

AGulev committed:
Update NDK 25 LTS (r25b) (#7240)

* fix script

* initial paths fixes

* use lld ref

* keep NDKs make file

* mk for libunwind fixes

* draft implementation of the unwinder using `_Unwind_Backtrace` and `_Unwind_GetIP`

* add `_Unwind_Backtrace` + `_Unwind_GetIP` implementation of the stack unwinding

* build luajit with ndk 25

* rebuild bullet for android

* rebuild protobuf for android

* Revert "draft implementation of the unwinder using `_Unwind_Backtrace` and `_Unwind_GetIP`"

This reverts commit e790b2591d4552b615622bca25f115608899b97f.

* fix LD arguments for protobuf on Android

* fix extender constants

* make sure this dmSDK uses NDK25b first

* don't need to link STL manually anymore

* use llvm-ar tools instead of GCC `ar`

* fix build_input for ndk25

* use NDK25 instead of NDK25b for enviroment variables

* use libunwind_ndk name

Jhonnyg committed:
Issue 6892: Shader include support (#6902)

* Moved compile code to ShaderUtil

* Use .glsl resource for include directive replacement

* Refactor

* Parse includes from cmd line

* WIP

* Move shader include compiler to separate class

* Editor support for shader includes work-in-progress

* Refactor include compiler

* Added shader include tests in bob

* Cleanup

* Fix tests

* Cleanup and two more tests

* Add .glsl template

* Use canonical path instead for resource resolving

* Minor simplification

* Refactor toProjectRelativePath to use relativize

* Remove unused

* Review fixes, use new regex

* Small cleanup

skip-ci

* Fix windows support

Co-authored-by: Mats Gisselson <mats.gisselson@gmail.com>

AGulev committed:
Update Android SDK (33.0.1 - Android 13 - API 33) (#7266)

* fix script

* initial paths fixes

* use lld ref

* keep NDKs make file

* mk for libunwind fixes

* draft implementation of the unwinder using `_Unwind_Backtrace` and `_Unwind_GetIP`

* add `_Unwind_Backtrace` + `_Unwind_GetIP` implementation of the stack unwinding

* build luajit with ndk 25

* rebuild bullet for android

* rebuild protobuf for android

* Revert "draft implementation of the unwinder using `_Unwind_Backtrace` and `_Unwind_GetIP`"

This reverts commit e790b2591d4552b615622bca25f115608899b97f.

* Upgraded android api and tools to 33 and 33.0.1 respectively in configuration files

* fix LD arguments for protobuf on Android

* fix extender constants

* make sure this dmSDK uses NDK25b first

* don't need to link STL manually anymore

* use llvm-ar tools instead of GCC `ar`

* fix build_input for ndk25

* use NDK25 instead of NDK25b for enviroment variables

* use libunwind_ndk name

* fix target version in meta.edn and SDK version in docs

* use android.jar from SDK 33.0.1

* update asan script

* Improvement in bob build-size report. Made it faster (#6654).

* Further reducing buid-size report execution time (#6654)

If many resources have the same content, this patch will
change the name of the resource displayed on the report.
It used to be the last one, now it's the first.

* Fixed broken unit tests (#6654)

ArchiveReader now works even if manifestFile property is not initialized.
That's the case in some unit tests.

* build input

* fix environment variable name

* use right path

* fix comment

Co-authored-by: Orestis Tsakiridis <otsakiridis@freemail.gr>

britzl committed:
Moved doc

--skip-ci

Jhonnyg committed:
Create buffer in runtime (#7214)

* Create texture resource in runtime

* Use same flow as DoGet for creating texture resource

* Error checking

* Fix tests

* Fix JS tests

* Save resource hash in collection and remove when collection is destroyed

* Add docstrings and cubemap uploading

* Add tests for resource.create_texture & resource.destroy

* Add more tests

* Review fixes

* Two more small fixes

* Add resource.set_atlas and resource.get_atlas

* Review fixes

* Refactor to use two-pass approach for parsing lua arguments

* Added documentation and some more validation

* Add unit tests

* Add flip flags

* Missed a comment..

* Add boilerplate resource.create_atlas fn

* Post review fixes

* Refactor resource atlas functions for code reuse

* Add tests for resource.create_atlas

* Experimental code

* Use hashes for vertexelement

* Use hashes in opengl

* Add missing opengl fn on win32

* Add tests and some crash fixes

* Minor pre-review touchups

* Fix mesh tests

* Hopefully fixed tests

* Missed bob changes

* Fixed compile error

* Add prebuilt mesh project

* Review fixes pt.1

* Change so that owership is always OWNER_RES

* Test changes

* Update prebuilt mesh data again

* Attempt to solve reference issue

* Clone buffer if creating resource from another buffer resource

* Move buffer copy into dmBuffer::Clone

vlaaad committed:
Make bob extract plugin zips (#7268)

Related to #3885

otsakir committed:
Removes/releases local JNI references in engine code (#6424)

...in all places where ThreadAttacher was used

britzl committed:
Update pull_request_template.md

Jhonnyg committed:
Support creating texture with buffer directly in resource.create  (#7285)

* Add compression_type param to resource.set_texture

* Create compressed texture from buffer as optional argument

* Add compression tests in gamesys

* Show help in texconvert tool

* Pre-review fixse

* Remove/move optional buffer in resource.create_texture, should be separate PR

* Remove/move again, remove comment about buffer in docstring for create_texture

* Re-add optional buffer as argument in resource.create_texture

* Add non-compressed support to texconvert

otsakir committed:
Bob debugging doc

Small piece of doc on how to debug Bob

JCash committed:
Factory dynamic prototype (#7264)

* Initial pipeline changes

* Revert "Initial pipeline changes"

This reverts commit 15321428d692a1baf5f6f832fc19d49334ebe5a2.

* Cleanup of api for dmResource::GetExtFromPath

* Initial version of setting collection factory prototype

* Added collectionfactory.set_prototype()

* Renamed resource function for collection factory

* Make the FactoryComponent private

* Added support for dynamic prorotype for game object factory

* Don't count factories with the "dynamic prototype" flag set

* Added unit test for counting resources when using the dynamic prototype setting

* Added prototype property to factory component

* Code cleanup

* Added dmScript::GetComponentFromLua to support being called from .script/.gui_script

* Test fix for waiting for the gameobjects to be created

* Updated editor test data with factory changes

britzl committed:
Document ports and services (#7296)

* Renamed file

* Create DEBUG_PORTS_AND_SERVICES.md

AGulev committed:
Do not scan`build` folder (#7316)

* add `build` as excluded folder

* exclude the whole `build` folder when bundle from the editor, not only `build/default`

AGulev committed:
Fix issue with components counter for cached resources (#7315)

* collect more info for time profiler

* use less info

* fix issue with task without inputs

* count components in gameobject

* cleanup counter code in gameobject

* add right inputs for collections

* take into account count of itstances

* get inputs from factories in embeded objects

* count embeded objects in collections

* copy values into collection

* make sure ogg counts right

* fix tests and exclude compcount_* from final bundle

* cleanup

* fix time profiler

* fix time report

* Take into account Dynamic Prototype

* add only unique inputs all the time

mikatuo committed:
Added missing types in documentation for "to" parameter in gui.animate and for "value" in go.property (#7328)

* Update docs for `gui.animate`

add number, quaternion to list of types of "to" property

* Update docs for go.property

boolean was missing from the list of accepted types

AGulev committed:
add more builders for unpack_ddf

JCash committed:
Bumped version to 1.4.3

AGulev committed:
use better defaults for screen orientation (#7329)

AGulev committed:
Fix issue when OGG file added to GO without component doesn't work in bundle (#7320)

* create component for ogg files as well as we do for wav

* fix comments

* fix tests

wookieejedi committed:
Update script_sys.cpp (#7331)

Jhonnyg committed:
Show error in bob when referencing missing script resource properties (#7343)

* Show error in bob when referencing missing script resource props

* Support empty resource properties

britzl committed:
Use min x and y scale for particle fx scale (#7336)

JCash committed:
Don't update the index buffer if the size is zero (#7338)

vlaaad committed:
Improve GUI modification performance by 15% (#7347)

User-facing changes: the editor is more performant when modifying highly-referenced GUI nodes

Related to #5447

wookieejedi committed:
Update template.emitter

vlaaad commented:
Many thanks for the contribution! I think this PR should target `editor-dev` instead of `dev`. Can you please change it so it targets that branch? That way, we can deliver the changes to our users sooner (i.e. as a new editor update after the merge instead of as a part of the 1.4.3 release in a month).

vlaaad committed:
Fail the build on invalid requires (#7350)

User-facing changes: the editor will now fail the build if a script file requires an existing lua module, but starts the `require` argument string with a slash.

Fixes #7348

JCash committed:
Use same class loader for bob as the bob plugins (#7349)

* Use same class loader for bob as the bob plugins

* review fix

britzl committed:
Added Void Scrappers and Monkey Mart (#7354)

Also stripped some metadata from existing images
